src/BlockchainTestsFiller/ValidBlocks/bcStateTests/selfdestructBalanceFiller.yml

```diff
@@ -0,0 +1,181 @@
+# Check the effects of selfDestruct on balance.
+# It should zero the balance twice:
+# 1. When it happens
+# 2. End of transaction
+
+selfdestructBalance:
+
+  genesisBlockHeader:
+    bloom: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+    coinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    difficulty: 0
+    extraData: 0x42
+    gasLimit: F00000000000
+    gasUsed: 0
+    mixHash: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+    nonce: 0x0102030405060708
+    number: 0
+    parentHash: 0x0000000000000000000000000000000000000000000000000000000000000000
+    receiptTrie: 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421
+    stateRoot: 0xf99eb1626cfa6db435c0836235942d7ccaa935f1ae247d3f1c21e495685f903a
+    timestamp: 0x54c98c81
+    transactionsTrie: 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421
+    uncleHash: 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347
+    baseFeePerGas: 10
+
+
+  _info:
+    comment: Ori Pomerantz     qbzzt1@gmail.com
+
+  pre:
+
+    # The "user" account
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: F000000000000000
+      code: 0x
+      nonce: 1
+      storage: {}
+
+    # Suicidal contract
+    000000000000000000000000000000000000dead:
+      balance: 10
+      code: |
+        :yul {
+          selfdestruct(0x1000)
```

Comment:
In the end this is much more readable then the suicide contract with complex selector logic in statetest

Reply:
Yes, but it can't check things such as "SELFDESTRUCT(dst1) the first time you're called, SELFDESTRUCT(dst2) the second time" without additional logic.